### PR TITLE
First go at typescript type definitions

### PR DESCRIPTION
**Pull request by pbadenski**
I started using this in our project. This is by no means complete and may contain some errors. But! It still helps immensely when adding custom integrations (if you use TypeScript that is).

Related to:
 - https://github.com/openzipkin/zipkin-js/issues/13
 - https://github.com/openzipkin/zipkin-js/issues/28

### CODE REVIEW

**Comment by adriancole** [comment]
looks nice. we should have something in the README and some tests to show this works (ex besides eyeballing which if the only option, that's ok)

**Comment by pbadenski** [comment]
Makes sense. I'll have a shot at it.

**Comment by pbadenski** [comment]
I added info to the README.

As for the example usage - for people who use TypeScript there's no setup, it's "you just use it" kind of thing. Only thing I can think of is - I could move **zipkin-instrumentation-redis** to TypeScript for the purpose of an example. Depends on how well this aligns with plans for **zipkin-js**.

P.S. I also fixed binary annotation type definition, based on my mess-up in #133

File: packages/zipkin/README.md [hunk 1]
```diff
@@ -3,6 +3,8 @@
 This is the core npm package for Zipkin. It contains the public API which is used by the various
 plugins (instrumentations and transports).
 
+We include TypeScript [definition file](index.d.ts) which you can also use a documentation.
```
Comment (adriancole) [inline packages/zipkin/README.md]: 'use a documentation' or did you mean use as documentation?

File: packages/zipkin/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,159 @@
+declare namespace zipkin {
```
Comment (adriancole) [inline packages/zipkin/index.d.ts]: Maybe make a comment when to change this? Ex whenever we change any file?

File: packages/zipkin/README.md (hunk 1 above)
Comment (pbadenski) [reply packages/zipkin/README.md]: I did mean "use as documentation"

File: packages/zipkin/index.d.ts (hunk 2 above)
Comment (pbadenski) [reply packages/zipkin/index.d.ts]: I will add a comment.

**Comment by pbadenski** [comment]
I understand your discomfort in this regard. Another idea - if tests were actually written in TypeScript, this would also help. But it does increase complexity. Unfortunately I'm not aware of other way to verify implementation matches TypeScript interfaces.

**Comment by adriancole** [comment]
going out at 0.9.0 thanks for all the help, @pbadenski!

**Pull request merged**